back/src/adapters/secondary/immersionOffer/laBonneBoite/HttpLaBonneBoiteAPI.ts

```diff
@@ -15,147 +9,54 @@ import {
-    );
-    return deduplicatedCompaniesProps.map(
+    const { responseBody } = await this.limiter.schedule(async () => {
+      const accessToken = await this.accessTokenGateway.getAccessToken(
+        `application_${this.poleEmploiClientId} api_labonneboitev1`,
+      );
+
+      return this.httpClient.getCompany({
+        headers: {
+          authorization: createAuthorization(accessToken.access_token),
+        },
+        queryParams: {
+          distance: MAX_DISTANCE_IN_KM,
+          longitude: searchParams.lon,
+          latitude: searchParams.lat,
+          page: 1,
+          page_size: MAX_PAGE_SIZE,
+          rome_codes: searchParams.rome,
+          sort: "distance",
+        },
+      });
+    });
+
+    return responseBody.companies.map(
       (props: LaBonneBoiteCompanyProps) => new LaBonneBoiteCompanyVO(props),
     );
   }
-  private async getCompanyResponse(
-    params: HttpGetLaBonneBoiteCompanyParams,
-  ): Promise<AxiosResponse<HttpGetLaBonneBoiteCompanyResponse>> {
-    return this.retryStrategy.apply(async () => {
-      try {
-        const axios = createAxiosInstance(logger);
-        const response = await this.rateLimiter.whenReady(async () => {
-          const accessToken = await this.accessTokenGateway.getAccessToken(
-            `application_${this.poleEmploiClientId} api_labonneboitev1`,
-          );
-          return axios.get(this.urlGetCompany, {
-            headers: {
-              Authorization: createAuthorization(accessToken.access_token),
-            },
-            timeout: secondsToMilliseconds(10),
-            params,
-          });
-        });
-        return response;
-      } catch (error: any) {
-        if (isRetryableError(logger, error)) throw new RetryableError(error);
-        logAxiosError(logger, error);
-        throw error;
-      }
-    });
-  }
 
-  private async recursivelyGetAllCompanies(
-    params: HttpGetLaBonneBoiteCompanyParams,
-    allCompanies: LaBonneBoiteCompanyProps[],
-  ): Promise<LaBonneBoiteCompanyProps[]> {
-    try {
-      const companyResponse = await this.getCompanyResponse(params);
-
-      const updatedAllCompanies = [
-        ...allCompanies,
-        ...companyResponse.data.companies,
-      ];
-
-      const wasLastPage =
-        companyResponse.data.companies_count === updatedAllCompanies.length;
-
-      return wasLastPage
-        ? updatedAllCompanies
-        : await this.recursivelyGetAllCompanies(
-            { ...params, page: params.page + 1 },
-            updatedAllCompanies,
-          );
-    } catch (error: any) {
-      logger.error(
-        "Error while recursively calling LBB API with params ",
-        params,
-        " : ",
-        error,
-      );
-      return allCompanies;
-    }
-  }
+  private limiter = new Bottleneck({
+    reservoir: lbbMaxQueryPerSeconds,
+    reservoirIncreaseInterval: 1000, // number of ms
+    reservoirIncreaseAmount: lbbMaxQueryPerSeconds,
+  });
```

Comment:
```suggestion
  private limiter = new Bottleneck({
    reservoir: lbbMaxQueryPerSeconds,
    reservoirIncreaseInterval: 1000, // number of ms
    reservoirIncreaseAmount: lbbMaxQueryPerSeconds,
    reservoirIncreaseMaximum: lbbMaxQueryPerSeconds
  });
```
C'est peut être pas nécessaire mais le test d'intégration lance les appels dès la création de l'adapter. J'ai peur que le réservoir continue d'augmenter au delà de `lbbMaxQueryPerSeconds`. Faudrait tester ce qu'il se passe si dans le test avec promise.all tu fais un setTimeout pour le lancer 2 secondes après le lancement du test.